samplesheets/tests/test_permissions_api_taskflow.py

```diff
@@ -74,3 +83,329 @@ def test_get_archive(self):
+    """Test permissions for IrodsRequestUpdateAPIView"""
+
+    def setUp(self):
+        super().setUp()
+        # Set up URLs
+        self.url_create = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        # Set up post data
+        self.post_data = {'path': self.path + '/', 'description': 'bla'}
+        self.update_data = {'path': self.path, 'description': 'Updated'}
+
+    def test_update(self):
+        """Test post() in IrodsRequestUpdateAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,  # Inherited owner
+            self.user_owner,
+            self.user_delegate,
+        ]
+        bad_users = [self.user_guest, self.user_no_roles, self.user_contributor]
+
+        with self.login(self.superuser):
+            self.client.post(self.url_create, self.post_data)
+            obj = IrodsDataRequest.objects.first()
+
+        self.url = reverse(
+            'samplesheets:api_irods_request_update',
+            kwargs={'irodsdatarequest': obj.sodar_uuid},
+        )
+        self.assert_response_api(
+            self.url, good_users, 200, method='POST', data=self.update_data
+        )
+        self.assert_response_api(
+            self.url, bad_users, 403, method='POST', data=self.update_data
+        )
+
+    @override_settings(PROJECTROLES_ALLOW_ANONYMOUS=True)
+    def test_update_anon(self):
+        """Test post() in IrodsRequestUpdateAPIView with anonymous access"""
+        self.project.set_public()
+        with self.login(self.superuser):
+            self.client.post(self.url_create, self.post_data)
+            obj = IrodsDataRequest.objects.first()
+
+        self.url = reverse(
+            'samplesheets:api_irods_request_update',
+            kwargs={'irodsdatarequest': obj.sodar_uuid},
+        )
+        self.assert_response_api(
+            self.url,
+            self.anonymous,
+            401,
+            method='POST',
+            data=self.update_data,
+        )
+
+
+class TestIrodsRequestDeleteAPIView(TestIrodsRequestAPIViewBase):
+    """Test permissions for IrodsRequestDeleteAPIView"""
+
+    def setUp(self):
+        super().setUp()
+        # Set up URLs
+        self.url_create = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        # Set up post data
+        self.post_data = {'path': self.path + '/', 'description': 'bla'}
+
+    def test_delete(self):
+        """Test delete() in IrodsRequestDeleteAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,  # Inherited owner
+            self.user_owner,
+            self.user_delegate,
+        ]
+        bad_users = [self.user_guest, self.user_no_roles, self.user_contributor]
+
+        for user in good_users:
+            with self.login(self.superuser):
```

Comment:
The thing is, that I need a uuid of a created request, but cleanup_method can not provide it(

Reply:
Not a problem :) Create UUID at the start of test or in the `setUp()` method, then set that UUID to the newly created object in the cleanup method. Ta-da, object with a static UUID :)

Reply:
That sounds quite good, but:
When I create a url object and then call the `assert_response_api()` method, then this url is being chnaged in cleanup_method. But the url, which is used for sending the requests (a local variable inside function) can not be changed that way (as in that case the whole function `assert_response_api()` have to be called once again, or somehow the local url variable changed to the url object, that I am creating in the beginning). 
So I spent quite time on that, but didn't find any proper way with cleanup_method(), which was better then a loop. Maybe I misunderstood you?

Reply:
I think my earlier explanation was not the best. You can see a recent example in this class, works fine :)

https://github.com/bihealth/sodar-server/blob/dev/samplesheets/tests/test_permissions.py#L1178

Reply:
We had a talk about this and I was wrong, in this case a built-in restriction in the SODAR Core implementation of this test feature does in fact prevent doing this. This will be improved in bihealth/sodar-core#1234.